src/clients/db/LabRepository.ts

```diff
@@ -1,12 +1,38 @@
 import Lab from 'model/Lab'
 import Repository from './Repository'
 import { labs } from '../../config/pouchdb'
+import SortRequest from './SortRequest'
 
+interface SearchContainer {
+  text: string
+  status: 'requested' | 'completed' | 'canceled' | 'all'
+  sortRequest: SortRequest
+}
 export class LabRepository extends Repository<Lab> {
   constructor() {
     super(labs)
-    labs.createIndex({
-      index: { fields: ['requestedOn'] },
+  }
+
+  async search(container: SearchContainer): Promise<Lab[]> {
+    const searchValue = { $regex: RegExp(container.text, 'i') }
+    const selector = {
+      $and: [
+        {
+          type: searchValue,
+        },
+        {
+          status: container.status,
+        },
+      ],
+      sorts: container.sortRequest.sorts,
+    }
+
+    if (container.status === 'all') {
+      selector.$and.splice(1, 1)
```

Comment:
Lets never add it to the `$and` array if `status === 'all'`. I fear that this code is easily broken when a new field is added or something gets removed.